### PR TITLE
Add EventBuilder implementation

### PR DESCRIPTION
**Pull request by buyaa-n**
Add EventBuilder implementation for new reflection emit project

Contributes to https://github.com/dotnet/runtime/issues/92975

### CODE REVIEW

File: src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/EventBuilderImpl.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Reflection.Metadata;
+
+namespace System.Reflection.Emit
+{
+    internal sealed class EventBuilderImpl : EventBuilder
+    {
+        private string _name;
```
Comment (steveharter) [inline src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/EventBuilderImpl.cs:11]: Can this and the next 3 fields be readonly?

File: src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs [hunk 2]
```diff
@@ -170,9 +171,10 @@ internal void AppendMetadata(MethodBodyStreamEncoder methodBodyEncoder)
                 }
 
                 WriteCustomAttributes(typeBuilder._customAttributes, typeHandle);
-                WriteProperties(typeBuilder);
                 WriteFields(typeBuilder);
                 WriteMethods(typeBuilder._methodDefinitions, genericParams, methodBodyEncoder);
+                WriteProperties(typeBuilder);
```
Comment (steveharter) [inline src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs:176]: Why the order change for properties? Does C# emit this way?

**Review by steveharter — APPROVED** [review]
Some questions, otherwise LGTM

File: src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs (hunk 2 above)
Comment (buyaa-n) [reply src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs:176]: nope, this order doesn't really matter as accessor methods handles are generated on type creation

File: src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/EventBuilderImpl.cs (hunk 1 above)
Comment (buyaa-n) [reply src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/EventBuilderImpl.cs:11]: yes, the can

**Pull request merged**